src/persist-txn/src/txn_read.rs

```diff
@@ -296,8 +338,50 @@ impl<T: Timestamp + Lattice + Codec64> TxnsRead<T> {
     }
 }
 
+/// Cancels an in-flight wait command when dropped, unless the given `tx` is
+/// yanked before that.
+struct CancelWaitOnDrop<T> {
```

Comment:
I don't really like this, and that we have to get a handle to the async runtime in `drop` ... 🙈 but I didn't see a better option for now.